airbyte-ci/connectors/pipelines/pipelines/contexts.py

```diff
@@ -476,6 +488,190 @@ async def __aexit__(
+            report_output_prefix (str): The S3 key to upload the test report to.
+            use_remote_secrets (bool, optional): Whether to download secrets for GSM or use the local secrets. Defaults to True.
+            gha_workflow_run_url (Optional[str], optional): URL to the github action workflow run. Only valid for CI run. Defaults to None.
+            dagger_logs_url (Optional[str], optional): URL to the dagger logs. Only valid for CI run. Defaults to None.
+            pipeline_start_timestamp (Optional[int], optional): Timestamp at which the pipeline started. Defaults to None.
+            ci_context (Optional[str], optional): Pull requests, workflow dispatch or nightly build. Defaults to None.
+            slack_webhook (Optional[str], optional): The slack webhook to send messages to. Defaults to None.
+            reporting_slack_channel (Optional[str], optional): The slack channel to send messages to. Defaults to None.
+            pull_request (PullRequest, optional): The pull request object if the pipeline was triggered by a pull request. Defaults to None.
+        """
+
+        self.pipeline_name = pipeline_name
+        self.cdk = cdk
+        self.use_remote_secrets = use_remote_secrets
+        self.report_output_prefix = report_output_prefix
+        self._secrets_dir = None
+        self._updated_secrets_dir = None
+        self.cdk_version = None
+        self.should_save_report = should_save_report
+
+        super().__init__(
+            pipeline_name=pipeline_name,
+            is_local=is_local,
+            git_branch=git_branch,
+            git_revision=git_revision,
+            gha_workflow_run_url=gha_workflow_run_url,
+            dagger_logs_url=dagger_logs_url,
+            pipeline_start_timestamp=pipeline_start_timestamp,
+            ci_context=ci_context,
+            slack_webhook=slack_webhook,
+            reporting_slack_channel=reporting_slack_channel,
+            pull_request=pull_request,
+            ci_report_bucket=ci_report_bucket,
+            ci_gcs_credentials=ci_gcs_credentials,
+            ci_git_user=ci_git_user,
+            ci_github_access_token=ci_github_access_token,
+        )
+
+    @property
+    def modified_files(self):
+        return self.cdk.modified_files
+
+    @property
+    def secrets_dir(self) -> Directory:  # noqa D102
+        return self._secrets_dir
+
+    @secrets_dir.setter
+    def secrets_dir(self, secrets_dir: Directory):  # noqa D102
+        self._secrets_dir = secrets_dir
+
+    @property
+    def updated_secrets_dir(self) -> Directory:  # noqa D102
+        return self._updated_secrets_dir
+
+    @updated_secrets_dir.setter
+    def updated_secrets_dir(self, updated_secrets_dir: Directory):  # noqa D102
+        self._updated_secrets_dir = updated_secrets_dir
+
+    @property
+    def should_save_updated_secrets(self) -> bool:  # noqa D102
+        return self.use_remote_secrets and self.updated_secrets_dir is not None
+
+
+    async def __aexit__(
```

Comment:
I don't get why you need to override `__aexit__`

Reply:
I think we should first try to make it use the default __aexit__ from its parent class and override if needed